cortex-m/src/delay.rs

```diff
@@ -75,62 +75,87 @@ impl Delay {
     }
 }
 
-impl DelayMs<u32> for Delay {
+#[cfg(feature = "eh0")]
+impl eh0::blocking::delay::DelayMs<u32> for Delay {
     #[inline]
     fn delay_ms(&mut self, ms: u32) {
         Delay::delay_ms(self, ms);
     }
 }
 
 // This is a workaround to allow `delay_ms(42)` construction without specifying a type.
-impl DelayMs<i32> for Delay {
+#[cfg(feature = "eh0")]
+impl eh0::blocking::delay::DelayMs<i32> for Delay {
     #[inline(always)]
     fn delay_ms(&mut self, ms: i32) {
         assert!(ms >= 0);
         Delay::delay_ms(self, ms as u32);
     }
 }
 
-impl DelayMs<u16> for Delay {
+#[cfg(feature = "eh0")]
+impl eh0::blocking::delay::DelayMs<u16> for Delay {
     #[inline(always)]
     fn delay_ms(&mut self, ms: u16) {
         Delay::delay_ms(self, u32::from(ms));
     }
 }
 
-impl DelayMs<u8> for Delay {
+#[cfg(feature = "eh0")]
+impl eh0::blocking::delay::DelayMs<u8> for Delay {
     #[inline(always)]
     fn delay_ms(&mut self, ms: u8) {
         Delay::delay_ms(self, u32::from(ms));
     }
 }
 
-impl DelayUs<u32> for Delay {
+#[cfg(feature = "eh0")]
+impl eh0::blocking::delay::DelayUs<u32> for Delay {
     #[inline]
     fn delay_us(&mut self, us: u32) {
         Delay::delay_us(self, us);
     }
 }
 
 // This is a workaround to allow `delay_us(42)` construction without specifying a type.
-impl DelayUs<i32> for Delay {
+#[cfg(feature = "eh0")]
+impl eh0::blocking::delay::DelayUs<i32> for Delay {
     #[inline(always)]
     fn delay_us(&mut self, us: i32) {
         assert!(us >= 0);
         Delay::delay_us(self, us as u32);
     }
 }
 
-impl DelayUs<u16> for Delay {
+#[cfg(feature = "eh0")]
+impl eh0::blocking::delay::DelayUs<u16> for Delay {
     #[inline(always)]
     fn delay_us(&mut self, us: u16) {
         Delay::delay_us(self, u32::from(us))
     }
 }
 
-impl DelayUs<u8> for Delay {
+#[cfg(feature = "eh0")]
+impl eh0::blocking::delay::DelayUs<u8> for Delay {
     #[inline(always)]
     fn delay_us(&mut self, us: u8) {
         Delay::delay_us(self, u32::from(us))
     }
 }
+
+impl DelayNs for Delay {
+    #[inline]
+    fn delay_ns(&mut self, ns: u32) {
+        Delay::delay_us(self, ns.saturating_add(999) / 1000)
```

Comment:
We don't have a 1 GHz system clock, so some rounding error seems reasonable. We should add a note about the rounding though.

Reply:
Yes, rounding is fine. However the API docs clearly specify that it's only allowed to round up:
"Pauses execution for at minimum ns nanoseconds. Pause can be longer if the implementation requires it due to precision/timing issues."

In rp2040-hal we do:
```
    fn delay_ns(&mut self, ns: u32) {
        // For now, just use microsecond delay, internally. Of course, this
        // might cause a much longer delay than necessary. So a more advanced
        // implementation would be desirable for sub-microsecond delays.
        let us = ns / 1000 + if ns % 1000 == 0 { 0 } else { 1 };
        // With rustc 1.73, this can be replaced by:
        // let us = ns.div_ceil(1000);
        self.delay_us_internal(us)
    }
```
I'd hope that the `/ 1000` and the `% 1000` can be optimized into a single division-with-remainder by the compiler, but I didn't verify that.

Reply:
Fixed.